awx/main/models/oauth.py

```diff
@@ -31,6 +31,13 @@ class Meta:
         editable=False,
         validators=[RegexValidator(DATA_URI_RE)],
     )
+    organization = models.ForeignKey(
+        'Organization',
+        related_name='applications',
+        help_text=_('Organization containing this application.'),
+        on_delete=models.CASCADE,
+        null=True,
+    )
```

Comment:
Do the tokens cascade delete from the application, and should they?

Reply:
They should because all tokens related to that application are useless if the application doesn't exist.  I just tried this and they don't cascade on delete like I would expect from this model..  Looking in to that now.